qiskit/circuit/_add_control.py

```diff
@@ -170,9 +170,6 @@ def control(
 
             apply_basic_controlled_gate(controlled_circ, gate, q_control, target)
 
-            if gate.definition is not None and gate.definition.global_phase and gate.name != "rz":
-                global_phase += gate.definition.global_phase
```

Comment:
This was an actual bug, which never got triggered because almost all of our equivalences do not update the global phase (and which got triggered after updating the definition for the ``ECRGate``).